### PR TITLE
Uniformed DataTypes to match the ones supported by kura protobuf.

### PR DESCRIPTION
**Pull request by MMaiero**
Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/TypedValue.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
  * @see TypedValues
  * @see BooleanValue
  * @see ByteArrayValue
- * @see ByteValue
+ * @see FloatValue
  * @see DoubleValue
  * @see IntegerValue
  * @see ShortValue
```
Comment (amitjoy) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/TypedValue.java]: Remove `ShortValue` as well

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java [hunk 2]
```diff
@@ -442,18 +442,15 @@ private void wrapValue(final AssetRecord assetRecord, final String userValue, fi
             if (BOOLEAN == dataType) {
                 value = TypedValues.newBooleanValue(Boolean.parseBoolean(userValue));
             }
-            if (BYTE == dataType) {
-                value = TypedValues.newByteValue(Byte.parseByte(userValue));
+            if (FLOAT == dataType) {
+                value = TypedValues.newFloatValue(Byte.parseByte(userValue));
```
Comment (amitjoy) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java]: `Float.parseFloat`

**Review by amitjoy — CHANGES_REQUESTED** [review]
Minor Change Requests

**Comment by MMaiero** [comment]
As stated in HSQLDB only DOUBLE is the result of a query. I've not found a simple way to keep floats after a query.
I think that, for the database, we must point out that in the documentation. For the rest of the components (asset, publisher, etc) everything seems fine.

**Comment by MMaiero** [comment]
Closes #1295

**Comment by MMaiero** [comment]
Closes #1294

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/FloatValue.java [hunk 3]
```diff
@@ -8,63 +8,62 @@
...
 
     /**
      * The actual contained value that will be represented as
      * {@link TypedValue}.
      */
-    private final byte value;
+    private final float value;
 
     /**
      * Instantiates a new byte value.
```
Comment (amitjoy) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/FloatValue.java]: new float value

**Review by amitjoy — CHANGES_REQUESTED** [review]

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/FloatValue.java (hunk 3 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/type/FloatValue.java]: Done

**Review by amitjoy — APPROVED** [review]

**Pull request merged**